readthedocs

MatthewSteen opened the pull request:
Fix #1. Changed the repo to public to add readthedocs. Note, the GraphHandler methods' docstrings aren't autodoc'd because they're decorated. https://buildingmotif.readthedocs.io/

haneslinger commented on docs/Makefile:
```diff
@@ -0,0 +1,20 @@
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
```
When I call this make file, I have to run `poetry run make ...`, As I'm using poetry with it's default setting `virtualenvs.create True` and no venv.

the fact your git ignoring `.venv` makes me thing you must be using poetry with `virtualenvs.create False` and within a venv shell. 

In Gabe's [makefile](https://github.com/NREL/BuildingMOTIF/pull/1/files) and the current README, commands include `poetry run`, assuming the default, non-venv configuration. We can have a dicussion about what we want to do, but ultimately, the docs and entrypoints should uniformly choose one or the other.

MatthewSteen replied:
Sounds good. We were using poetry _with_ a virtual environment in tasty, so that's what I did here. I don't have a preference either way, however no venv seems cleaner/simpler.

TShapinsky replied:
from the team meeting yesterday we decided that without venv is cleaner and also that these kind of scripts should not assume the user is already in the virtual environment and as such should use `poetry run <cmd>`. This has the added benefit that it should work in either case (in virtual env or not).

haneslinger reviewed (changes requested):
Looking good! Some small comments.

MatthewSteen commented on buildingmotif/GraphHandler.py:
```diff
@@ -66,7 +66,7 @@ def create_graph(self, identifier: str, graph: Optional[Graph] = None) -> Graph:
 
     @_with_db_open
-    def get_all_graph_identifiers(self) -> list[str]:
+    def get_all_graph_identifiers(self) -> list:
```
I was getting a pytest error.

TShapinsky replied:
We decided yesterday that we're going to require types in all function signatures. So I'd like to get to the root of why this is failing for you.

TShapinsky replied:
@haneslinger part of the problem could be using `list` instead of the `List` object from `typing`

MatthewSteen replied:
https://stackoverflow.com/a/31905868/4228267
> Starting with Python 3.9, you can use `list[str]` as a type annotation, which doesn't require importing anything.

MatthewSteen commented on docs/source/buildingmotif.rst:
```diff
@@ -0,0 +1,9 @@
+buildingmotif
```
Done, thanks!

MatthewSteen commented:
Thanks for the review @haneslinger. I think I've addressed your comments, so let me know if there are any other changes you'd like. Especially for the function annotation I changed for the pytest error. If there aren't any other changes, please resolve the comments and merge.

haneslinger commented on README.md:
```diff
@@ -1,20 +1,31 @@
-# BuildingMotif [![codecov](https://codecov.io/gh/haneslinger/BuildingMotif/branch/main/graph/badge.svg?token=2SNN5HPOHL)](https://codecov.io/gh/haneslinger/BuildingMotif)
+# BuildingMOTIF [![Documentation Status](https://readthedocs.org/projects/buildingmotif/badge/?version=latest)](https://buildingmotif.readthedocs.io/en/latest/?badge=latest) [![codecov](https://codecov.io/gh/haneslinger/BuildingMotif/branch/main/graph/badge.svg?token=2SNN5HPOHL)](https://codecov.io/gh/haneslinger/BuildingMotif)
```
Love the badge.

haneslinger reviewed (approved):
Great Work! ✨
